VisualPinball.Unity/VisualPinball.Unity.Editor/Utils/TreeView/TreeElement.cs

```diff
@@ -0,0 +1,75 @@
+using System;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace VisualPinball.Unity.Editor.Utils.TreeView
+{
+	/// <summary>
+	/// TreeElement is the base class for creating TreeModel which will be used by TreeViewWithTreeModel, you'll have to inherit from it.
+	/// It provides base properties to be inserted & identified in a TreeView (Id, Name, Depth)
+	/// It hosts also its hierarchical information.
+	/// </summary>
+	public abstract class TreeElement
+	{
+		/// <summary>
+		/// Unique ID of the TreeElement within the Tree structure
+		/// </summary>
+		public int Id { get; set; }

```

Comment:
I'm wondering... do the IDs have to be sequential? If not, maybe using the game object's [instance ID](https://docs.unity3d.com/ScriptReference/Object.GetInstanceID.html) would be easier?

Reply:
I'm already using gameobject instance Id for items treeelements, only layers treeelements are using indexes, they're not related to gameobjects

Reply:
Ah, you're right, sorry I missed that!